sources/engine/Stride.Engine/Engine/Startup/UsageExample/MyGameApp.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Stride.Engine.Startup;
+using Stride.Rendering.Lights;
+
+namespace MyGame;
+
+// The OS-specific project (what would be in the project MyGame.Windows.csproj)
+// would still do minimal work.
+// (I have only checked windows, I assume the other OSes are similar)
+class MyGameApp
+{
+    static void Main()
+    {
+        using (var game = new MyGameSetup().SetUpGame())
+        {
+            game.Run();
+        }
+    }
+}
+
+// The game specific project (what would be in the project MyGame.csproj)
+// would have the setup class that is called by the different OS-specific
+// projects. However, at this point, the different parts are not hooked
+// up to the engine yet, so it would not do anything significant.
+public class MyGameSetup : IGameSetup
+{
+    public IStrideApplication SetUpGame()
+    {
+        var gameBuilder = StrideApplication.CreateBuilder();
+        return gameBuilder.Build();
+    }
+}
+
+// Later on, the setup can compose the application parts following
+// the patter of how ASP Core does it:
+public class MyGameSetupLater : IGameSetup
+{
+    public IStrideApplication SetUpGame()
+    {
+        var gameBuilder = StrideApplication.CreateBuilder();
+
+        // Use configuration values
+        var meta = gameBuilder.Configuration.GetSection("Meta");
+        var version = meta["Version"];
+
+        if (gameBuilder.Environment.IsDevelopment())
+        {
+            // set up logging
+            // The game imports Microsoft.Extensions.Logging.Console,
+            // But the engine only needs Microsoft.Extensions.Logging.
+            // (This example file is added to the engine so now it also needs console)
+            gameBuilder.Logging.AddConsole();
+        }
+
+        // Register dependencies
+        gameBuilder.Services.AddSingleton<Dependency>();
```

Comment:
I read on a blog post once that FOSS has 2 disadvantages compared to closed source software (also a bunch of advantages for sure, but the point is:):
- Work that is less "attractive" is less likely to get done
- People are much less likely to sit down in a corner of the office and draw funny shapes on a whiteboard.

IMHO using `Scoped` really needs the latter to add value. For example, when I discussed this with Manio, he was a proponent of a scope being _a frame_.

---

To give you my considerations from my game where I'm using this, for what it's worth:

It felt too much like "If you have a hammer, everything looks like a nail". It was certainly built with Sdk.Web in mind, and, well, think of the bread and butter for ASP, your usual B2B application: It's not out of the ordinary to have 1 million LOC, but a lot of it is edge-case stuff, like reports that are only generated once a month anyway. And a single web request only ever needs a very small part of that. Not having everything in memory constantly directly saves the company money by allowing cheaper server hosting with less RAM. `Scoped` just makes so much sense for such a web application.

But for a game engine, you're working with a whole different situation. Much more RAM available on average, and if it becomes a bottleneck, it's probably because of the assets, and not an overwhelming amount of code.

Honestly, I haven't encountered a hard reason for `Transient` in my game yet, either. Effectively, my transient services are always referenced by services that are singletons in the end, so they effectively all have singleton lifetime - with the difference that multiple instances are created if it's used in multiple places. The only practical advantage I currently have is documentation a la "this class does not have state that I need to keep alive" (but that also kinda just because I decided to use it that way, for now).

Ugh, I've already written way more than I assumed this would be :D Anyway, my advice would be to just not even touch `Scoped` at first. If you want to do something with it, you can still add that later, the infrastructure is there - but it will become a whole own discussion.

---

And yeah, that kinda also goes for the `EntityProcessor`/`ScriptComponent` part, where I agree with the general sentiment that that's more of a future step. Once it is, indeed, in place - but also, sufficiently used throughout the codebase - some things will probably just "melt away" and produce a clearer picture how these things can work together.

I have a feeling already that the mechanism that's essentially activating and deactivating `EntityProcessor`s should not be the same as the IoC container; that those are 2 distinct responsibilities... but that also strikes me as a question where you need some funny shapes on a whiteboard first.